bdb-ip.md

```diff
@@ -0,0 +1,437 @@
+# Improvement proposals for HTTP API
+
+
+Currently, there is two milestones grouping issues for upcoming HTTP API
+endpoints:
+
+- [Changes that break the existing HTTP API of BDB
+  v0.11](https://github.com/bigchaindb/bigchaindb/milestone/11)
+- [Changes that are additions to the exiting HTTP API of BDB
+  v0.11](https://github.com/bigchaindb/bigchaindb/milestone/9)
+
+After releasing BDB v1.0, breaking changes to the external APIs are not
+permitted anymore. Additions to the APIs are however always allowed.
+
+To resolve the outstanding issues, we'll hence limit
+the scope of this document to the already existing endpoints.
+
+
+### First task
+
+Please check the two listed milestones above for the correctness of categorized
+issues.
+
+
+### Methodology
+
+This document will go address each of the issues in the [breaking
+changes milestone](https://github.com/bigchaindb/bigchaindb/milestone/11) and
+discuss possible solutions to them. Once there is a consensus on the solutions,
+we can implement them into BDB core (mainly HTTP API), py and js driver and
+the BDB CLI.
+
+
+### Issues in milestone
+
+#### [Inconsistent naming of short-hand ids](https://github.com/bigchaindb/bigchaindb/issues/1134)
+
+Current naming of ids according to [HTTP API documentation
+(master)](https://docs.bigchaindb.com/projects/server/en/master/http-client-server-api.html)
+and the [Websocket Event Stream API
+(master)](https://docs.bigchaindb.com/projects/server/en/master/websocket-event-stream-api.html):
+
+- Transaction id:
+    - `tx_id` in query parameters and Event Stream API
+    - `txid` in `transaction.inputs.fulfills`
+    - `id` in transaction payload
+    - `id` in `/assets?search={text_search}` (this is also the asset's id!)
+    - `ID` in prose text
+- Asset id:
+    - `asset_id` in query parameters and Event Stream API
+    - `id` in `transaction.asset` payload
+    - `id` in `/assets?search={text_search}` (this is also the transaction's id!)
+- Fulfillment id:
```

Comment:
I thought about this as well.

Ultimately, we'll have to go with whatever json structure Interledger proposes, meaning that we might just want to ignore everything within a condition or fulfillment.